File: 412-fizz-buzz/fizz-buzz.swift
class Solution {

    private func getItem(for itemIndex: Int) -> String {

        if itemIndex % 3 == 0 && itemIndex % 5 == 0 {
            return "FizzBuzz"
        }

        if itemIndex % 3 == 0 {
            return "Fizz"
        }

        if itemIndex % 5 == 0 {
            return "Buzz"
        }

        return String(itemIndex)
    }

    func fizzBuzz(_ n: Int) -> [String] {
        
        return (1...n).map { self.getItem(for: $0) }
    }
}

File: 412-fizz-buzz/fizz-buzz.java
import java.util.stream.IntStream;

class Solution {
    public List<String> fizzBuzz(int n) {

        return IntStream.rangeClosed(1, n)
                .mapToObj((i) -> {
                    if (i % 3 == 0 && i % 5 == 0) {
                        return "FizzBuzz";
                    }

                    if (i % 3 == 0) {
                        return "Fizz";
                    }

                    if (i % 5 == 0) {
                        return "Buzz";
                    }

                    return Integer.toString(i);
                    
                }).collect(Collectors.toList());
    }
}